airbyte-webapp/src/views/Connector/ServiceForm/components/EditControls.tsx

```diff
@@ -62,25 +66,34 @@ const EditControls: React.FC<IProps> = ({
     <>
       {renderStatusMessage()}
       <Controls>
-        <div className={styles.buttonsContainer}>
-          <Button type="submit" disabled={isSubmitting || !dirty || Object.keys(unfinishedFlows).length > 0}>
-            <FormattedMessage id="form.saveChangesAndTest" />
-          </Button>
-          <Button
-            className={styles.cancelButton}
-            type="button"
-            variant="secondary"
-            disabled={isSubmitting || !dirty}
-            onClick={onCancelClick}
-          >
-            <FormattedMessage id="form.cancel" />
-          </Button>
-        </div>
-        {onRetestClick && (
-          <Button type="button" onClick={onRetestClick} disabled={!isValid}>
-            <FormattedMessage id={`form.${formType}Retest`} />
+        {onDelete && (
+          <Button variant="danger" onClick={onDeleteButtonClick} data-id="open-delete-modal">
+            <FormattedMessage id={`tables.${formType}Delete`} />
           </Button>
         )}
+        <div className={styles.buttonsContainer}>
+          {dirty && (
+            <>
+              <Button type="submit" disabled={isSubmitting || !dirty || Object.keys(unfinishedFlows).length > 0}>
+                <FormattedMessage id="form.saveChangesAndTest" />
+              </Button>
+              <Button
+                className={styles.cancelButton}
+                type="button"
+                variant="secondary"
+                disabled={isSubmitting || !dirty}
+                onClick={onCancelClick}
+              >
+                <FormattedMessage id="form.cancel" />
+              </Button>
```

Comment:
Can we swap the order of these two buttons, and change the `form.saveChangesAndTest` text to be `Save & Test`?

Those two changes would bring this in line with the design on Figma: 

<img width="641" alt="Screen Shot 2022-10-20 at 5 16 47 PM" src="https://user-images.githubusercontent.com/22731524/197082811-4ae80ab8-2579-4615-b59e-db20488ba9b6.png">

Reply:
fixed this one